contribution_guidelines.md

```diff
@@ -0,0 +1,51 @@
+# pyBDY contribution guidelines
+
+## Set up a development conda environment
+
+Before making any changes to the code, it is recommended to create a new conda environment for development:
+
+```bash
+conda env create -n pybdy-dev python=3.9
+conda env update -n pybdy-dev -f environment.yml
+conda activate pybdy-dev
+```
+
+## Create a branch to work on an issue
+
+If you don't have write permissions on NOC-MSM/pyBDY, fork the repository and work from your own fork.
+
+1. Open a GitHub issue or pick one already opened
+1. Create a branch that references the issue number and gives a summary of the changes that will be made:  `git branch issue-103-remove-pynemo-traces`
+1. Switch to that branch (i.e., update HEAD to set that branch as the current branch): `git checkout issue-103-remove-pynemo-traces`
+1. Make your changes to the code
+
+## Pre-push steps
+
+Before pushing to GitHub, run the following commands:
+
+1. Update conda environment: `make conda-env-update`
+1. Install this package: `pip install -e .`
+1. Sync with the latest [template](https://github.com/ecmwf-projects/cookiecutter-conda-package) (optional): `make template-update`
+1. Install pre-commit hooks: `pre-commit install`
+1. Run quality assurance checks: `make qa`
+1. Run tests: `make unit-tests`
+1. Run the static type checker (currently not working for pyBDY): `make type-check`
+1. Build the documentation (optional, see [Sphinx tutorial](https://www.sphinx-doc.org/en/master/tutorial/)): `make docs-build`
+
+## Push to GitHub
+
+1. Add changes to the staging area with `git add src/pybdy/file_change.py`
+1. Commit changes: `git commit -m "#103 Remove pynemo traces"`
+1. As hooks are run on every commit because before we executed `pre-commit install`, they may flag problems with the code and/or change some of its parts, thus preventing the commit attempt to be successful. If all pre-commit hooks passed, jump to step 7.
+1. Solve any issue flagged by the pre-commit hooks.
+1. Update already stag files executing `git add -u`.
+1. Commit changes once again `git commit -m "#103 Remove pynemo traces"`.
+1. Finally, push changes to GitHub and set the remote as upstream with `git push --set-upstream origin issue-103-remove-pynemo-traces`.
+
+## Make a pull request
+
```

Comment:
Should there be a "merge master into branch" step here, or maybe at the start of the Pre-push steps?

Reply:
I think it is better to merge a branch into master (and solve any conflicts that may arise at that point) than the master into a branch. In this way, we are sure that the upstream changes don't break a PR. 

Maybe something along the lines of

```
git checkout master
git pull
```
before creating a branch could help to ensure that whatever branch is created, it starts from the current HEAD of the upstream master.

Reply:
My concern would be more about a PR breaking the master, than the other way around.
(With the workflow ordering you propose, I do see how the qa and unit testing would be applied as the conflicts would be resolved on the GitHub server?)

BTW I'm happy to try your proposal and learn by doing. I consider this issue "resolved"

Reply:
OK, I got your point now.

So yes, if we want the developers to be responsible for resolving any potential conflicts, then they should merge master into branch before pushing. If the PR has no conflicts, I think there is no point in merging as it just introduces a redundant commit. 

I will adapt the guidelines to introduce these ideas.